File: Product-of-Array-Except-Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        left = [1]*len(nums)
        right = [1]*len(nums)
        result= []
        for i in range(1,len(nums)):
            left[i] = left[i-1]*nums[i-1]
            right[len(nums)-1-i] = right[len(nums)-i]*nums[len(nums)-i]

        for i in range(len(nums)):
            result.append(left[i]*right[i])
        return result
 
     

File: Product-of-Array-Except-Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] products = new int[nums.length];
        int[] L = new int[nums.length];
        int[] R = new int[nums.length];
        L[0] = 1;
        R[nums.length-1] = 1;
        for(int i = 1; i < nums.length; i++)
        {
            L[i] = L[i-1]*nums[i-1];
            R[nums.length-1-i] =R[nums.length-i]*nums[nums.length-i];
        }
        
        for(int i = 0; i < nums.length; i++)
        {
            products[i]  = L[i]*R[i];
        }     
        return products;     
    }
}
        
